android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/worker/MeasureReportWorker.kt

```diff
@@ -0,0 +1,174 @@
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.quest.ui.report.measure.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingWorkPolicy
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.search.search
+import com.google.android.fhir.workflow.FhirOperator
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.temporal.ChronoUnit
+import java.util.Calendar
+import java.util.Date
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.withContext
+import org.hl7.fhir.r4.model.Measure
+import org.hl7.fhir.r4.model.MeasureReport
+import org.smartregister.fhircore.engine.configuration.ConfigType
+import org.smartregister.fhircore.engine.configuration.ConfigurationRegistry
+import org.smartregister.fhircore.engine.configuration.app.ApplicationConfiguration
+import org.smartregister.fhircore.engine.data.local.DefaultRepository
+import org.smartregister.fhircore.engine.util.DefaultDispatcherProvider
+import org.smartregister.fhircore.engine.util.extension.SDF_YYYY_MM_DD
+import org.smartregister.fhircore.engine.util.extension.alreadyGeneratedMeasureReports
+import org.smartregister.fhircore.engine.util.extension.firstDayOfMonth
+import org.smartregister.fhircore.engine.util.extension.formatDate
+import org.smartregister.fhircore.engine.util.extension.lastDayOfMonth
+import org.smartregister.fhircore.engine.util.extension.loadCqlLibraryBundle
+import org.smartregister.fhircore.engine.util.extension.parseDate
+import org.smartregister.fhircore.engine.util.extension.plusMonths
+import org.smartregister.fhircore.engine.util.extension.today
+import org.smartregister.fhircore.quest.ui.report.measure.MeasureReportViewModel
+import timber.log.Timber
+
+@HiltWorker
+class MeasureReportWorker
+@AssistedInject
+constructor(
+  @Assisted val appContext: Context,
+  @Assisted workerParams: WorkerParameters,
+  val defaultRepository: DefaultRepository,
+  val configurationRegistry: ConfigurationRegistry,
+  val dispatcherProvider: DefaultDispatcherProvider,
+  val fhirOperator: FhirOperator,
+  val fhirEngine: FhirEngine
+) : CoroutineWorker(appContext, workerParams) {
+
+  val applicationConfiguration: ApplicationConfiguration by lazy {
+    configurationRegistry.retrieveConfiguration(ConfigType.Application, configId = "app/debug")
```

Comment:
```suggestion
    configurationRegistry.retrieveConfiguration(ConfigType.Application)
```

Another reason is hard coding this will crash the app for apps with different appIds